drivers/clk/renesas/rzg2l-cpg.c

```diff
@@ -550,27 +594,44 @@ rzg2l_cpg_sd_mux_clk_register(const struct cpg_core_clk *core,
 
 static unsigned long
 rzg2l_cpg_get_foutpostdiv_rate(struct rzg2l_pll5_param *params,
-			       unsigned long rate)
+			       unsigned long rate,
+			       struct rzg2l_cpg_priv *priv)
 {
 	unsigned long foutpostdiv_rate;
+	unsigned int lanes;
+	unsigned int bpp;
 	unsigned long hsclk;
+	struct device_node *np = NULL;
+	u32 dsi_div;
 
 	/*
-	 * Relationship between hsclk and vclk must follow
-	 * vclk * bpp = hsclk * 8 * lanes
-	 * where vclk: video clock (Hz)
-	 *       bpp: video pixel bit depth
-	 *       hsclk: DSI HS Byte clock frequency (Hz)
-	 *       lanes: number of data lanes
-	 * Currently default value:
-	 *		 bpp is 24
-	 *		 lanes is 1
-	 *		 => hsclk = 3 * vclk
-	 * intin = (hsclk * 16 * refdiv * postdiv1 * postdiv2) / EXTAL_FREQ
-	 * fracin = ((hsclk * 16 * refdiv * postdiv1 * postdiv2) % EXTAL_FREQ) << 24) / EXTAL_FREQ
-	 */
-	hsclk = 3 * rate;
+	* Get DSI status and update lanes and bpp values.
+	* If the DSI node is found, retrieve the values.
+	* Otherwise, use default values.
+	*/
+	if(get_node_status("dsi", &np)){
```

Comment:
@long-trinh-rvc .

- Add two new fields for the panel. Then you can use them for the condition.
- Update the panel with the new fields in the panel node.

Reply:
Or you can move the calculation before the du timing.

[drivers/gpu/drm/renesas/rz-du/rzg2l_du_crtc.c#L64](https://github.com/vudangRVC/linux-rz-sst/blob/styhead/rz-sbc-review-dsi/drivers/gpu/drm/renesas/rz-du/rzg2l_du_crtc.c#L64)